src/naksu/ui.go

```diff
@@ -342,133 +378,155 @@ func populateBackupCombobox(backupMedia map[string]string, combobox *ui.Combobox
+		for _, buttonRule := range buttonRules {
+			if buttonRule.enable {
+				buttonRule.element.Enable()
+			} else {
+				buttonRule.element.Disable()
 			}
 		}
-	}()
+
+		for _, comboboxRule := range comboboxRules {
+			if comboboxRule.enable {
+				comboboxRule.element.Enable()
+			} else {
+				comboboxRule.element.Disable()
+			}
+		}
+	})
+}
+
+func getStatusForMainUIStatusHandler() (bool, bool, bool) {
+	// Query box installation status
+	boxInstalled, boxInstalledErr := box.Installed()
+	if boxInstalledErr != nil {
+		log.Debug(fmt.Sprintf("Could not query whether VM is installed: %v", boxInstalledErr))
+	}
+
+	boxInstalled = (boxInstalledErr == nil) && boxInstalled
+
+	// Query box running status
+	boxRunning, boxRunningErr := box.Running()
+	if boxRunningErr != nil {
+		log.Debug(fmt.Sprintf("Could not query whether VM is running: %v", boxRunningErr))
+	}
+
+	boxRunning = (boxRunningErr == nil) && boxRunning
+
+	// Query network (internet) connection status
+	netAvailable := network.CheckIfNetworkAvailable()
+
+	return boxInstalled, boxRunning, netAvailable
 }
 
 // checkAbittiUpdate checks
 // 1) if currently installed box is Abitti
 // 2) and there is a new version available
 func checkAbittiUpdate() (bool, string) {
-	abittiUpdate := false
 	availAbittiVersion := ""
 
-	currentBoxType := box.GetType()
 	currentBoxVersion := box.GetVersion()
 
-	if boxversion.GetVagrantBoxTypeIsAbitti(currentBoxType) {
-		_, availBoxVersion, errAvail := boxversion.GetVagrantBoxAvailVersionDetails()
-		if errAvail == nil && currentBoxVersion != availBoxVersion {
-			abittiUpdate = true
-			availAbittiVersion = availBoxVersion
+	boxInstalled, boxInstalledErr := box.Installed()
+
+	if (boxInstalledErr == nil && !boxInstalled) || box.TypeIsAbitti() {
+		var errAvail error
+
+		availAbittiVersion, errAvail = cloud.GetAvailableVersion(constants.AbittiVersionURL)
+		if errAvail == nil && currentBoxVersion != availAbittiVersion {
+			return true, availAbittiVersion
 		}
 	}
 
-	return abittiUpdate, availAbittiVersion
+	return false, availAbittiVersion
 }
 
 // updateStartButtonLabel updates label for start button depending on the
 // installed VM style. If there is no box installed the default label is set.
 func updateStartButtonLabel() {
 	go func() {
-		boxTypeString := boxversion.GetVagrantBoxType(box.GetType())
+		boxTypeString := box.GetTypeLegend()
 		ui.QueueMain(func() {
 			if boxTypeString == "-" {
 				buttonStartServer.SetText(xlate.Get("Start Exam Server"))
 			} else {
-				buttonStartServer.SetText(fmt.Sprintf(xlate.Get("Start %s"), boxTypeString))
+				buttonStartServer.SetText(xlate.Get("Start %s", boxTypeString))
 			}
 		})
 	}()
 }
 
-// updateVagrantBoxAvailLabel updates UI "update available" label if the currently
+// updateBoxAvailLabel updates UI "update available" label if the currently
 // installed box is Abitti and there is new version available
-func updateVagrantBoxAvailLabel() {
+func updateBoxAvailLabel() {
```

Comment:
Onko tarpeen käyttää näitä lyhenteitä? Voisiko olla `updateBoxAvailabilityLabel` tms?

Reply:
:heavy_check_mark: